.github/workflows/build.yml

```diff
@@ -0,0 +1,75 @@
+name: Test and Build Workflow
+# This workflow is triggered on pull requests to main branch
+on:
+  pull_request:
+    branches:
+      - main
+  push:
+    branches:
+      - main
+
+jobs:
+  build:
+    # Job name
+    name: Build Replication plugin
+    runs-on: ubuntu-latest
+    steps:
+      # This step uses the setup-java Github action: https://github.com/actions/setup-java
+      - name: Set Up JDK 14
+        uses: actions/setup-java@v1
+        with:
+          java-version: 14
+      # dependencies: OpenSearch
+      - name: Checkout OpenSearch
+        uses: actions/checkout@v2
+        with:
+          # TODO: Point to opensearch repo after the patch changes have been merged.
+          repository: 'saikaranam-amazon/OpenSearch'
+          #repository: 'opensearch-project/OpenSearch'
+          path: OpenSearch
+          ref: 'karanas-os-replication'
+      - name: Build OpenSearch
+        working-directory: ./OpenSearch
+        run: |
+          ./gradlew publishToMavenLocal -Dbuild.snapshot=false
+      # dependencies: common-utils
+      - name: Checkout common-utils
+        uses: actions/checkout@v2
+        with:
+          ref: 'main'
+          repository: 'opensearch-project/common-utils'
+          path: common-utils
+      - name: Build common-utils
+        working-directory: ./common-utils
+        run: ./gradlew publishToMavenLocal -Dbuild.snapshot=false -Dopensearch.version=1.1.0
+      # This step uses the checkout Github action: https://github.com/actions/checkout
+      - name: Checkout Branch
+        uses: actions/checkout@v2
+      - name: Build and run Replication tests
+        run: |
+          ./gradlew clean release
+      - name: Upload failed logs
+        uses: actions/upload-artifact@v2
+        if: failure()
+        with:
+          name: logs
+          path: |
+            build/testclusters/integTest-*/logs/*
+            build/testclusters/leaderCluster-*/logs/*
+            build/testclusters/followCluster-*/logs/*
+      - name: Create Artifact Path
+        run: |
+          pwd
+          mkdir -p cross-cluster-replication-artifacts
+          cp ./build/distributions/*.zip cross-cluster-replication-artifacts
+          pwd
+      - name: Uploads coverage
+        uses: codecov/codecov-action@v1.2.1
+        with:
+          token: ${{ secrets.CODECOV_TOKEN }}
+      # This step uses the upload-artifact Github action: https://github.com/actions/upload-artifact
```

Comment:
Do you need this in CI? I think you should enable CI on all branches without this part and make a CD workflow if you really want one. Unlike ODFE those artifacts aren't going to be used starting 1.1 because we build from the top.

Reply:
removed the `upload-artifact` step